File: leetcode/go/83_remove_duplicates_from_sorted_list.go
package main 

type ListNode struct {
	Val  int
	Next *ListNode
}

func deleteDuplicates(head *ListNode) *ListNode {
	p := head
	for p != nil {
		d := p.Next
		for d != nil && p.Val == d.Val {
			d = d.Next
		}
		p.Next = d
		p = d
	}
	return head
}

// github.com/p-nerd (Shihab Mahamud)
// Wednesday, April 17, 2024 | 05:32:06 PM (+6)

File: leetcode/cpp/83_remove_duplicates_from_sorted_list.cpp
#include <bits/stdc++.h>
using namespace std;

struct ListNode
{
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr)
    {
    }
    ListNode(int x) : val(x), next(nullptr)
    {
    }
    ListNode(int x, ListNode *next) : val(x), next(next)
    {
    }
};

class Solution
{
  public:
    ListNode *deleteDuplicates(ListNode *head)
    {
        ListNode *p = head;
        while (p != nullptr)
        {
            ListNode *d = p->next;
            while (d != nullptr && p->val == d->val)
            {
                d = d->next;
            }
            p->next = d;
            p = d;
        }
        return head;
    }
};

int main()
{

    return 0;
};
// github.com/p-nerd (Shihab Mahamud)
// Wednesday, April 17, 2024 | 05:31:52 PM (+6)
